SingularityUI/app/components/common/Application.jsx

```diff
@@ -32,23 +32,23 @@ class Application extends Component {
     }, DISMISS_TASK_LAG_NOFICATION_DURATION_IN_MS);
   }
 
-  notifyLag(maxTaskLag) {
+  notifyLateRequests(listLateTasks) {
     const { canShowTaskLagNotification: canNotify } = this.state;
-    const shouldNotify = maxTaskLag >= MAX_TASK_LAG_NOTIFICATION_THRESHOLD_IN_MS;
+    const lateRequests = Utils.getListOfUniqueRequestsFromListOfTasks(listLateTasks)
+    const shouldNotify = lateRequests.length >= MAX_LATE_REQUESTS;
     if (canNotify && shouldNotify) {
       Messenger().error({
         onClickClose: this.dismissTaskLagNotification,
         message: `
           Singularity is experiencing some delays. The team has already been
-          notified. (Max task lag: ${Utils.duration(maxTaskLag)})
+          notified. (Number of distinct late requests: ${lateRequests.length})
         `,
       });
     }
   }
 
   render() {
-    this.notifyLag(this.props.maxTaskLag);
-
+    this.notifyLateRequests(this.props.listLateTasks);
```

Comment:
I see this was here before, but this call should probably go in `componentDidMount`

Reply:
Since we want to make this call whenever the listOfLateTasks prop changes, I'll put it in componentDidUpdate